src/librustc/error_codes.rs

```diff
@@ -1896,6 +1896,49 @@ fn foo<'a>(x: &'a i32, y: &i32) -> &'a i32 {
 ```
 "##,
 
+E0623: r##"
+A lifetime didn't match what was expected.
+
+Erroneous code example:
+
+```compile_fail,E0623
+struct Foo<'a> {
+    x: &'a isize,
+}
+
+fn bar<'short, 'long>(c: Foo<'short>, l: &'long isize) {
+    let _: Foo<'long> = c; // error!
+}
+```
+
+In this example, we tried to set a value with an incompatible lifetime to
+another one (`'long` != `'short`). We can solve this issue in two different
```

Comment:
I'd be tempted to say "is unrelated to" rather than just `!=`:

```suggestion
another one (`'long` is unrelated to `'short`). We can solve this issue in two different
```